### PR TITLE
IDULS-2540 add delete phone number gen-not

### PR DESCRIPTION
**Pull request by AnjBB72**
**PR Description:**

Adding general email notification placeholders and templates for a user deleting a phone number from their profile.
This relates to new feature by IDULS team in Identity value stream.

PR going into a feature branch (each required notification is being done as a separate piece of work) which will then only be merged when all notifications completed.

### CODE REVIEW

File: collections/business/general-notifications/event-general-notifications/phone-number-deleted/event-general-notifications.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "eventClassName": "com.backbase.dbs.user.manager.event.spec.v2.DeletePhoneNumberSecNotEvent",
+  "id": "phone-number-deleted",
+  "eventBusinessName": "Phone Number Deleted",
+  "recipientType": "internal",
+  "extractor": "#this['userInternalId']",
+  "generalNotifications": [
+    {
+      "id": "phone-number-deleted",
+      "name": "Phone number deleted",
```
Comment (amorimluzia) [inline collections/business/general-notifications/event-general-notifications/phone-number-deleted/event-general-notifications.json:10]: @AnjBB72 seems this GN does not have preferences, right? Please add userPreferenceEnabled field here. E.g:
"userPreferenceEnabled": false,

Example: https://github.com/Backbase/engagements-data/blob/master/collections/retail/general-notifications/event-general-notifications/message-received/event-general-notifications.json

File: collections/retail/general-notifications/event-general-notifications/phone-number-deleted/event-general-notifications.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+  "eventClassName": "com.backbase.dbs.user.manager.event.spec.v2.DeletePhoneNumberSecNotEvent",
+  "id": "phone-number-deleted",
+  "eventBusinessName": "Phone Number Deleted",
+  "recipientType": "internal",
+  "extractor": "#this['userInternalId']",
+  "generalNotifications": [
+    {
+      "id": "phone-number-deleted",
+      "name": "Phone number deleted",
```
Comment (amorimluzia) [inline collections/retail/general-notifications/event-general-notifications/phone-number-deleted/event-general-notifications.json:10]: @AnjBB72 seems this GN does not have preferences, right? Please add userPreferenceEnabled field here. E.g:
"userPreferenceEnabled": false,

Example: https://github.com/Backbase/engagements-data/blob/master/collections/retail/general-notifications/event-general-notifications/message-received/event-general-notifications.json

File: collections/business/general-notifications/templates/phone-number-deleted-email/template.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+  "name":"phone-number-deleted-email",
+  "description":"Email template for phone number deleted notification",
+  "title":"Phone number deleted",
+  "engagement": "general-notification",
+  "channel": "email",
+  "placeholderGroups":["phoneNumberDeleted"]
+}
```
Comment (amorimluzia) [inline collections/business/general-notifications/templates/phone-number-deleted-email/template.json:8]: @AnjBB72 I can see you are using placeholders from user profile on the template as well, so here you should update the last line:
 "placeholderGroups":["phoneNumberDeleted", "userProfile"]

File: collections/retail/general-notifications/templates/phone-number-deleted-email/template.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "name":"phone-number-deleted-email",
+  "description":"Email template for phone number deleted notification",
+  "title":"Phone number deleted",
+  "engagement": "general-notification",
+  "channel": "email",
+  "placeholderGroups":["phoneNumberDeleted"]
+}
```
Comment (amorimluzia) [inline collections/retail/general-notifications/templates/phone-number-deleted-email/template.json:8]: @AnjBB72 I can see you are using placeholders from user profile on the template as well, so here you should update the last line:
 "placeholderGroups":["phoneNumberDeleted", "userProfile"]

File: collections/business/general-notifications/event-general-notifications/phone-number-deleted/event-general-notifications.json (hunk 1 above)
Comment (AnjBB72) [reply collections/business/general-notifications/event-general-notifications/phone-number-deleted/event-general-notifications.json:10]: Thanks @amorimluzia , I will get that updated

File: collections/business/general-notifications/templates/phone-number-deleted-email/template.json (hunk 3 above)
Comment (AnjBB72) [reply collections/business/general-notifications/templates/phone-number-deleted-email/template.json:8]: Thanks, I will get that updated.

**Review by amorimluzia — APPROVED** [review]

File: collections/retail/general-notifications/templates/phone-number-deleted-email/template.json (hunk 4 above)
Comment (amorimluzia) [reply collections/retail/general-notifications/templates/phone-number-deleted-email/template.json:8]: hey @AnjBB72 is userProfile not user-profile

File: collections/business/general-notifications/templates/phone-number-deleted-email/template.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+  "name":"phone-number-deleted-email",
+  "description":"Email template for phone number deleted notification",
+  "title":"Phone number deleted",
+  "engagement": "general-notification",
+  "channel": "email",
+  "placeholderGroups":["phoneNumberDeleted", "user-profile"]
+}
```
Comment (amorimluzia) [inline collections/business/general-notifications/templates/phone-number-deleted-email/template.json:8]: hey @AnjBB72 is userProfile not user-profile

File: collections/retail/general-notifications/templates/phone-number-deleted-email/template.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
+  "name":"phone-number-deleted-email",
+  "description":"Email template for phone number deleted notification",
+  "title":"Phone number deleted",
+  "engagement": "general-notification",
+  "channel": "email",
+  "placeholderGroups":["phoneNumberDeleted", "user-profile"]
```
Comment (amorimluzia) [inline collections/retail/general-notifications/templates/phone-number-deleted-email/template.json:7]: hey @AnjBB72 is userProfile not user-profile

**Review by amorimluzia — APPROVED** [review]

File: collections/business/general-notifications/event-general-notifications/phone-number-deleted/event-general-notifications.json [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+      "deliveryOptions": [
+        {
+          "channel": "email",
+          "channelSettings": "/phone-number-deleted/email/email_channel-settings.json",
+          "isEnabledByDefault": true,
+          "isManageableByUser": false,
+          "engagementTemplates": [
+            {
+              "path": "/phone-number-deleted/email/en_US/engagement-template.json",
+              "templateName": "phone-number-deleted-email",
```
Comment (eacarvalho) [inline collections/business/general-notifications/event-general-notifications/phone-number-deleted/event-general-notifications.json:21]: Hi 👋, I was reviewing another pull request about the same topic in another repository and all the comments apply to this one too, please can you take a look?

TL;DR: For email we don't have manageability via Engage App thus we don't need to inform templateName, template.json, content-schema.json and placeholders.json. Moreover there is an embedded way to specify the html file directly here :)

**Review by eacarvalho — CHANGES_REQUESTED** [review]

**Comment by afsinka-backbase** [comment]
Hello, is destination branch correct?

**Comment by AnjBB72** [comment]
> Hello, is destination branch correct?

@afsinka-backbase - Yes, we are doing a load of notification work for User Manager and Password reset so will have a few tickets being raised - we want them on feature branch first as our work on the services won't be going in until Feb so will look to merge the feature branch at the same time.

**Comment by afsinka-backbase** [comment]
Could you please drop a message in #s-engagements in Slack if/when you need a review, otherwise the team may not realise these PRs? Thanks!
